addons/mrp_byproduct/tests/test_mrp_byproduct.py

```diff
@@ -0,0 +1,72 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+from openerp.tests import common
+
+
+class TestMrpByProduct(common.TransactionCase):
+
+    def setUp(self):
+        super(TestMrpByProduct, self).setUp()
+        self.MrpProduction = self.env['mrp.production']
+        self.MrpBom = self.env['mrp.bom']
+        self.Move = self.env['stock.move']
+        self.MrpProductProduce = self.env['mrp.product.produce']
+        self.product_28 = self.env.ref('product.product_product_28')
+        self.product_33 = self.env.ref('product.product_product_33')
+        self.product_uom = self.env.ref('product.product_uom_unit')
+        self.location_src = self.env.ref('stock.stock_location_stock')
+        self.mrp_action = self.env.ref('mrp.menu_mrp_production_action')
```

Comment:
No need to recordset just need ***id*** then get directly ***id*** , why do first get ***recordset*** and then assign that recordset's ***id*** do it like : 
```
self.product_33_id = self.ref('product.product_product_33')
self.product_uom_id = self.ref('product.product_uom_unit')
self.location_src_id = self.ref('stock.stock_location_stock')
self.mrp_action_id = self.ref('mrp.menu_mrp_production_action')
```